framework/include/limiters/SOULimiter.h

```diff
@@ -47,18 +47,25 @@ class SOULimiter : public Limiter<T>
    * @param fi_elem_is_upwind Boolean flag indicating if the current element is upwind.
    * @return The computed flux limiting ratio.
    */
-  T limit(const T & phi_upwind,
-          const T & phi_downwind,
+  T limit(const T & /* phi_upwind */,
+          const T & /* phi_downwind */,
           const VectorValue<T> * grad_phi_upwind,
-          const VectorValue<T> * grad_phi_downwind,
+          const VectorValue<T> * /* grad_phi_downwind */,
           const RealVectorValue & dCD,
           const Real & max_value,
           const Real & min_value,
```

Comment:
I think that's the only thing I'd try to change back and It does not need to be done now. 
Approved

Reply:
Thanks. Note that SOU does not depend on the downwind gradients. The current implementation is correct

Reply:
yes it's correct. It's just T vs Real that maybe we can do something about. Nothing that affects results

Reply:
You mean for `max_value` and `min_value`? If so, it does make sense that they are real numbers. Limiting with the maximum value as a vector or ADReal does not really make sense imo